iot/iot-device-registry-base/src/main/java/io/enmasse/iot/registry/tenant/KubernetesTenantInformationService.java

```diff
@@ -2,56 +2,93 @@
  */
-
 package io.enmasse.iot.registry.tenant;
 
-import static io.enmasse.iot.registry.tenant.TenantInformation.of;
 import static io.vertx.core.Future.failedFuture;
 import static io.vertx.core.Future.succeededFuture;
 import static java.net.HttpURLConnection.HTTP_INTERNAL_ERROR;
+import static java.net.HttpURLConnection.HTTP_NOT_FOUND;
 
+import java.net.HttpURLConnection;
 import java.util.Optional;
 
 import org.eclipse.hono.client.ClientErrorException;
 import org.eclipse.hono.client.ServerErrorException;
+import org.eclipse.hono.deviceregistry.service.tenant.TenantInformationService;
+import org.eclipse.hono.deviceregistry.service.tenant.TenantKey;
+import org.eclipse.hono.service.management.OperationResult;
+import org.eclipse.hono.service.management.Result;
 import org.eclipse.hono.service.management.tenant.Tenant;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 import org.springframework.stereotype.Component;
 
 import com.fasterxml.jackson.core.JsonProcessingException;
 import com.fasterxml.jackson.databind.ObjectMapper;
 
 import io.enmasse.iot.model.v1.IoTProject;
 import io.enmasse.iot.service.base.AbstractProjectBasedService;
-import io.opentracing.SpanContext;
+import io.opentracing.Span;
 import io.vertx.core.Future;
 
 @Component
 public class KubernetesTenantInformationService extends AbstractProjectBasedService implements TenantInformationService {
 
+    private static final Logger log = LoggerFactory.getLogger(KubernetesTenantInformationService.class);
+
     private ObjectMapper mapper = new ObjectMapper();
 
     @Override
-    public Future<TenantInformation> tenantExists(final String tenantName, final int notFoundStatusCode, final SpanContext span) {
+    public Future<Result<TenantKey>> tenantExists(String tenantId, Span span) {
+
+        return getProject(tenantId)
+                .flatMap(project -> {
+                            if (!project.isEmpty()) {
+                                return validateTenant(project, tenantId, HTTP_NOT_FOUND)
+                                        .compose(tenant -> Future.succeededFuture(
+                                                OperationResult.ok(
+                                                        HttpURLConnection.HTTP_OK,
+                                                        TenantKey.from(tenant.getId(), tenant.getName()),
+                                                        Optional.empty(),
+                                                        Optional.empty())
+                                                )
+                                        );
+                            } else {
+                                return Future.succeededFuture(Result.from(HTTP_NOT_FOUND));
+                            }
+                        }
+                );
+    }
+
+    public Future<TenantInformation> tenantExists(final String tenantName, final int notFoundStatusCode, final Span span) {
```

Comment:
It is hard for me to grasp the difference of the two `tenantExists` methods. I think Javadoc would help.